src/as-suggested.c

```diff
@@ -0,0 +1,214 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2016 Lucas Moura <lucas.moura128@gmail.com>
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * SECTION: as-suggested
+ * @short_description: Descibe which componentes were responsible for
+ *					   a given package recommendation.
+ * @include: appstream.h
+ */
+
+#include "config.h"
+
+#include "as-suggested.h"
+
+typedef struct
+{
+	AsSuggestedKind	kind;
+	GPtrArray		*cpts_id; /* of string */
```

Comment:
On second thought, I also wonder if a GHashTable isn't a better idea here, since that will prevent us from having duplicate entries...